integration/token/fungible/views/swap.go

```diff
@@ -184,5 +204,26 @@ func (t *SwapResponderView) Call(context view.Context) (interface{}, error) {
 	assert.NoError(err, "failed to retrieve vault status for transaction [%s]", tx.ID())
 	assert.Equal(network.Valid, vc, "transaction [%s] should be in valid state", tx.ID())
 
+	// Check that the tokens are or are not in the vault
+	qe := vault.TokenVault().QueryEngine()
+	outputs, err := tx.Outputs()
+	assert.NoError(err, "failed to retrieve outputs")
+	for _, output := range outputs.Outputs() {
+		tokenID := output.ID(tx.ID())
```

Comment:
See my comment below regarding ID. This line confuses me a bit.
Why not just:
```go
tokenID := &token2.ID{TxId: tx.ID, Index: output.Index}
```
Not sure if we need a constructor function for that